### PR TITLE
RC indicators and delegations

### LINKED ISSUE #781: RC Indicator
Show indicator around user menu so we can inform level of remaining resources and possibly add some warning when running low. hiveblocks.com/@demo shows remaining number of comments/votes etc. Having similar estimations are great to establish some threshold.

### PR DESCRIPTION
**Pull request by feruzm**
Adds RC indicator on profile
Adds RC delegation operations
Delegations in and out list, detailed info about RC stats

Fixes #781

### CODE REVIEW

**Comment by Adesojisouljay** [comment]
### What does this PR?
This PR provides a feature that allows users to delegate resource credit to another user, it also allows users to update or delete existing delegations. 

Users can see more information about their resource credits by clicking on the Rc progress bar in user menu. Those information are available in the modal that pops up. Outgoing and incoming are clickable, when clicked, a modal consisting of incoming and outgoing delegations list respectively pops up.

### Screenshot/Video
[View Rc actions on loom](https://www.loom.com/share/cf1ccad1a67c404a89c898c762257c0d)
[View UI responsiveness on loom](https://www.loom.com/share/26e03541ce864fa5af8aa02bc415569a)

### Github issue
[find issue here](https://github.com/ecency/ecency-vision/issues/1096)

**Comment by feruzm** [comment]
@Adesojisouljay left comments on first loom video, please check.

**Comment by feruzm** [comment]
@Adesojisouljay couple issues remaining:

- updating delegation has faulty logic, when I try to update delegation, it opens dialog but amount changing keeps next button disabled until To username is modified again.
- deleting delegation should ask for confirmation before submitting transaction, just like here: https://github.com/ecency/ecency-vision/pull/1174
- RC modal only shows when you click progress bar, but let's make it open when clicking on text/label as well.
- Have one more option in Modal that says or shows or links to FAQ page with specific description to explain what RC is in more details.
- Fix delete delegation it is using `delegateRCKc` function which is wrong, should use `delegateRC` plainly because it is posting operation.
- Clean up code console.logs and unnecessary `delegateRCKc` and `delegateRCHot` functions
- `info` text can be removed from modal Title and label 
image

File: src/common/components/rc-info/index.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            console.log(availableResourceCredit)
+            // const formatRc: any = rcFormatter(Number(availableResourceCredit));
+            setresourceCredit(availableResourceCredit);
+            const inComing: any = r.map((a: any) => Number(a.received_delegated_rc));
+            const formatIncoming = rcFormatter(inComing);
+            setReceivedDelegation(formatIncoming);
+          })
+    }, []);
+
+    const rcFormatter = (num: number) => {
```
Comment (feruzm) [inline src/common/components/rc-info/index.tsx:39]: @Adesojisouljay I have moved this function to helpers, please merge development branch into this and use function from helpers. Reference: https://github.com/ecency/ecency-vision/pull/1194

**Comment by Adesojisouljay** [comment]
All noted and would be rounded up, thanks 🙏

**Comment by feruzm** [comment]
leaving here for reference, suggestion to improve UI modal

**Comment by Adesojisouljay** [comment]
Noted,  thanks🙏

**Pull request merged**